jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1073,6 +1079,22 @@ private HttpFields takeTrailers()
             return trailers;
         }
 
+        @Override
+        public void write(Content.Chunk chunk, Callback callback)
+        {
+            if (chunk instanceof Trailers trailersChunk)
+            {
+                ResponseHttpFields trailers = _request.getHttpChannel()._responseTrailers;
+                if (trailers != null)
+                    trailers.add(trailersChunk.getTrailers());
+                callback.succeeded();
```

Comment:
Don't we still need to write here?
```suggestion
                Response.super.write(true, callback);
```

Reply:
Done.